common/modules/gcp-project/main.tf

```diff
@@ -178,6 +178,7 @@ data "google_iam_policy" "combined" {
 
     members = [
       "${local.service_accounts}",
+      "serviceAccount:${google_project.project.number}@cloudbuild.gserviceaccount.com",
```

Comment:
I can not find which permissions `gcloud compute images export` needs for what. All that I can find is in the doc:
https://cloud.google.com/sdk/gcloud/reference/compute/images/export

And also in the errors that are raised in the logs when something fails:

https://console.cloud.google.com/logs/viewer?interval=JUMP_TO_TIME&project=gpii-gcp-stg&authuser=1&organizationId=247149361674&minLogLevel=0&expandAll=false&timestamp=2019-08-05T16%3A48%3A24.000000000Z&customFacets&limitCustomFacetWidth=false&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.location%3D%22us-central1%22%0Aresource.labels.project_id%3D%22gpii-gcp-stg%22%0Aresource.labels.pod_name%3D%22backup-exporter-1565034300-jzcbb%22%0Aresource.labels.cluster_name%3D%22k8s-cluster%22%0Aresource.labels.container_name%3D%22backup-container%22%0Aresource.labels.namespace_name%3D%22backup-exporter%22%0Atimestamp%3D%222019-08-05T19%3A45%3A13.853317678Z%22%0AinsertId%3D%22zp48fzah6zyne0y7x%22&scrollTimestamp=2019-08-05T19%3A46%3A15.930421706Z&dateRangeStart=2019-08-05T15%3A48%3A24.000Z&dateRangeUnbound=forwardInTime

Where we can see that the command failed because the `gke-cluster-pod-bckp-exporter@gpii-gcp-stg.iam.gserviceaccount.com` was not able to grant the privileges to the cloudbuild SA:
```
INFO: Adding [serviceAccount:774516236737@cloudbuild.gserviceaccount.com] to [roles/iam.serviceAccountTokenCreator
```

Reply:
> I can not find which permissions

@amatas the role itself (`roles/iam.serviceAccountUser`) is fine, but it should not be applied on the whole project, allowing Cloud build to use all service accounts in the project, but instead it should only be granted on individual service account (`gke-cluster-pod-bckp-exporter@..`), using the `google_service_account_iam_policy`  Terraform resource, and therefore allowing Cloud build to use only that particular account.

Reply:
I guess I explained wrongly. Cloud build does not need the role (roles/iam.serviceAccountUser) to use the SA (gke-cluster-pod-bckp-exporter@..).

The (`gke-cluster-pod-bckp-exporter@..`) SA tries to set the IAM role (roles/iam.serviceAccountUser) at project level for the `cloud build SA` when the `gcloud` command is fired if it is not already set, but it fails because the (`gke-cluster-pod-bckp-exporter@..`) SA does not have `setIamPolicy` permission.

I guess that the `cloud build SA` needs to use another SA of the project, but I do not know what it is.